app/models/tweet.rb

```diff
@@ -7,9 +7,11 @@ class Tweet < ApplicationRecord
 
   def generate_hashtags
     content.scan(/#\w+/).each do |tag|
-      hashtag = Hashtag.new(tag: tag)
-      hashtag.save
-      self.hashtags << hashtag
+      if not Hashtag.exists?(tag: tag)
```

Comment:
Consider using this type of thing instead? https://apidock.com/rails/ActiveRecord/Base/create_or_update